stl/src/atomic_wait.cpp

```diff
@@ -0,0 +1,235 @@
+    constexpr std::size_t _Wait_table_size       = 1 << _Wait_table_size_power;
+    constexpr std::size_t _Wait_table_index_mask = _Wait_table_size - 1;
+
+#pragma warning(push)
+#pragma warning(disable : 4324) // structure was padded due to alignment specifier
+    struct alignas(std::hardware_destructive_interference_size) _Wait_table_entry {
+        // Arbitraty variable to wait/notify on if target wariable is not proper atomic for that
+        // Size is largest of lock-free to make aliasing problem into hypothetical
+        std::atomic<std::uint64_t> _Counter;
+
+        CONDITION_VARIABLE _Condition = CONDITION_VARIABLE_INIT;
+        SRWLOCK _Lock                 = SRWLOCK_INIT;
+    };
+#pragma warning(pop)
+
+    _Wait_table_entry& _Atomic_wait_table_entry(const void* const _Storage) noexcept {
+        static _Wait_table_entry wait_table[_Wait_table_size];
+        auto index = reinterpret_cast<std::uintptr_t>(_Storage);
+        index ^= index >> (_Wait_table_size_power * 2);
+        index ^= index >> _Wait_table_size_power;
+        return wait_table[index & _Wait_table_index_mask];
+    }
+
+    enum _Atomic_spin_phase : std::size_t {
+        _Atomic_wait_phase_mask            = 0x0000'000F,
+        _Atomic_spin_value_mask            = 0xFFFF'FFF0,
+        _Atomic_spin_value_step            = 0x0000'0010,
+        _Atomic_wait_phase_init_spin_count = 0x0000'0000,
+        _Atomic_wait_phase_spin            = 0x0000'0002,
+        _Atomic_wait_phase_wait            = 0x0000'0001,
+    };
+
+    static_assert(_Atomic_unwait_needed == _Atomic_wait_phase_wait);
+
+    inline std::size_t _Atomic_get_spin_count() noexcept {
+        static std::size_t constexpr uninitialized_spin_count = (std::numeric_limits<std::size_t>::max)();
+        static std::atomic<std::size_t> atomic_spin_count     = uninitialized_spin_count;
+        std::size_t result                                    = atomic_spin_count.load(std::memory_order_relaxed);
+        if (result == uninitialized_spin_count) {
+            result = (std::thread::hardware_concurrency() == 1 ? 0 : 10'000) * _Atomic_spin_value_step;
+            atomic_spin_count.store(result, std::memory_order_relaxed);
+
+            // Make sure other thread is likely to get this,
+            // as we've done kernel call for that.
+            std::atomic_thread_fence(std::memory_order_seq_cst);
+        }
+        return result;
+    }
+
+    void _Atomic_wait_fallback(const void* const _Storage, std::size_t& _Wait_context) noexcept {
+        switch (_Wait_context & _Atomic_wait_phase_mask) {
+        case _Atomic_wait_phase_init_spin_count: {
+            _Wait_context = _Atomic_wait_phase_spin | _Atomic_get_spin_count();
+            [[fallthrough]];
+        }
+
+        case _Atomic_wait_phase_spin: {
+            if ((_Wait_context & _Atomic_spin_value_mask) > 0) {
+                _Wait_context -= _Atomic_spin_value_step;
+                YieldProcessor();

```

Comment:
@StephanTLavavej , I'm wondering why in `_YIELD_PROCESSOR()` defined in `xatomic.h` there's [no `_mm_pause` for x86](https://github.com/microsoft/STL/blame/master/stl/inc/xatomic.h#L31).
This Windows API version of `YieldProcessor()` inserts both x86 `pause` instruction and ARM `yield` instruction